pkg/machinery/resources/block/volume_lifecycle.go

```diff
@@ -0,0 +1,58 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package block
+
+import (
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/resource/meta"
+	"github.com/cosi-project/runtime/pkg/resource/protobuf"
+	"github.com/cosi-project/runtime/pkg/resource/typed"
+
+	"github.com/siderolabs/talos/pkg/machinery/proto"
+)
+
+// VolumeLifecycleType is type of VolumeLifecycle resource.
+const VolumeLifecycleType = resource.Type("VolumeLifecycles.kubernetes.talos.dev")
```

Comment:
Shouldn't it be `block`?

```suggestion
const VolumeLifecycleType = resource.Type("VolumeLifecycles.block.talos.dev")
```